### PR TITLE
Variable Server - var_send_once and integration test

### PR DESCRIPTION
**Pull request by jdeans289**
Makes several changes relating to the variable server:
- Adds the send_once command and message type, which allows a user to request a variable to be sent immediately and only once (intended to replace the var_add, var_send, var_clear idiom that is commonly used for this purpose)
- Minor refactoring of variable server internals to reduce repeated code
- Adds SIM_test_varserv to integration test to test basic variable server functionality
- Changes graphics client for SIM_billiards to use var_send_once as an example of intended use

### CODE REVIEW

File: trick_source/sim_services/VariableServer/VariableServerThread_copy_sim_data.cpp [hunk 1]
```diff
@@ -92,14 +93,11 @@ int Trick::VariableServerThread::copy_sim_data() {
...
+        if (cyclical) {
+            var_data_staged = true;
+            packets_copied++ ;
+        }
 
         pthread_mutex_unlock(&copy_mutex) ;
     }
-
-
-    return (0) ;
```
Comment (spfennell) [inline trick_source/sim_services/VariableServer/VariableServerThread_copy_sim_data.cpp:102]: We need this return 0. Double checked the build and we now get a warning from this.

File: include/trick/VariableServerThread.hh [hunk 2]
```diff
@@ -426,9 +449,19 @@ namespace Trick {
             int transmit_file(std::string file_name);
 
             /**
-             @brief Called by write_data to write data to socket in var_binary format.
+             @brief Called by write_data to write given variables to socket in var_binary format.
+            */
+            int write_binary_data( int Start, char *buf1, const std::vector<VariableReference *>& givenVars, VS_MESSAGE_TYPE messageType);
```
Comment (spfennell) [inline include/trick/VariableServerThread.hh:454]: I know Trick is inconsistent with variable names, but let's go ahead and call ```givenVars``` ```given_vars``` unless there's a good reason to use camel case here.

File: test/SIM_test_varserv/models/varserv/src/VS_tests.cpp [hunk 3]
```diff
@@ -33,112 +33,168 @@ int VSTest::strcmp_IgnoringWhiteSpace(const char* s1, const char* s2) {
     }
 }
 
-
+// Not really sure what this is supposed to be testing, should probably just remove
```
Comment (spfennell) [inline test/SIM_test_varserv/models/varserv/src/VS_tests.cpp:36]: Yes, you can remove anything in the VS tests that you don't think is meaningful.

File: trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp [hunk 4]
```diff
@@ -46,17 +46,18 @@ REF2* Trick::VariableServerThread::make_error_ref(std::string in_name) {
     return new_ref;
 }
 
-int Trick::VariableServerThread::var_add(std::string in_name) {
-
-    VariableReference * new_var ;
+Trick::VariableReference* Trick::VariableServerThread::create_var_reference(std::string in_name) {
```
[the remark this answers is not in the document]
Comment (spfennell) [reply trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp:49]: Although I'm reading about compilers and now I think we can probably just let the compiler inline it if it wants to.

File: trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp [hunk 5]
```diff
@@ -93,6 +99,36 @@ int Trick::VariableServerThread::var_add(std::string var_name, std::string units
     return(0) ;
 }
 
+// TODO: This should go somewhere else probably?
```
Comment (spfennell) [inline trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp:102]: It's used only here, personally I'd leave it. Maybe add a comment saying ```//Helper function for var_send_once``` so we can see that it's just for this file at a glance

File: trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (jdeans289) [reply trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp:49]: Left as is

File: trick_source/sim_services/VariableServer/VariableServerThread_copy_sim_data.cpp (hunk 1 above)
Comment (jdeans289) [reply trick_source/sim_services/VariableServer/VariableServerThread_copy_sim_data.cpp:102]: Fixed. I'm not sure why I took that out, probably an accident.

File: include/trick/VariableServerThread.hh (hunk 2 above)
Comment (jdeans289) [reply include/trick/VariableServerThread.hh]: Fixed this and some other camel case variables that I saw

File: trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp (hunk 5 above)
Comment (jdeans289) [reply trick_source/sim_services/VariableServer/VariableServerThread_commands.cpp]: Added

File: test/SIM_test_varserv/models/varserv/src/VS_tests.cpp (hunk 3 above)
Comment (jdeans289) [reply test/SIM_test_varserv/models/varserv/src/VS_tests.cpp]: Removed this and some other commented out code in VS test

**Comment by jdeans289** [comment]
closes #1323 
closes #1332 
closes #1311

**Pull request merged**